tests/h2-tests/tests/server.rs

```diff
@@ -296,10 +296,10 @@ async fn push_request_against_concurrency() {
             .await;
         client.recv_frame(frames::data(2, &b""[..]).eos()).await;
         client
-            .recv_frame(frames::headers(1).response(200).eos())
+            .recv_frame(frames::headers(4).response(200).eos())
             .await;
         client
-            .recv_frame(frames::headers(4).response(200).eos())
+            .recv_frame(frames::headers(1).response(200).eos())
```

Comment:
@jwilm this was an interesting change in behavior... I'm guessing this is because of the `push_front` vs `push_back` that you added?

Reply:
That sounds a likely cause. There's another option with how the pending_open/pending_send queues are managed where we could prefer to pop data frames before popping new streams. I think that would actually be a bit closer to the old behavior, but it would make the `poll_complete` implementation more complex.